server/WebAPI/Startup.cs

```diff
@@ -13,108 +13,117 @@
-                                        .AllowAnyHeader()
-                                        .AllowAnyMethod()
-                                        .AllowCredentials()
-                                        .WithOrigins("http://localhost:8080");
-                                  });
-            });
-
-            services.AddDbContext<RepositoryContext>(options => 
-                options.UseSqlServer(Configuration.GetConnectionString("mssqlConnection")));
-            services.AddSingleton(Configuration);
-            services.AddScoped<UserRepository>();
-            services.AddScoped<UserService>();
-            services.AddSingleton<AuthService>();
-            services.AddScoped<TripRepository>();
-            services.AddScoped<TripService>();
-            services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
-                .AddJwtBearer(x =>
-            {
-                x.RequireHttpsMetadata = false;
-                x.SaveToken = true;
-                x.TokenValidationParameters = new TokenValidationParameters
-                {
-                    ValidateIssuerSigningKey = true,
-                    IssuerSigningKey = new SymmetricSecurityKey(Encoding.ASCII.GetBytes(Configuration["Jwt:Key"])),
-                    ValidateIssuer = true,
-                    ValidIssuer = Configuration["Jwt:Issuer"],
-                    ValidateAudience = true,
-                    ValidAudience = Configuration["Jwt:Audience"]
-                };
-            });
-            services.AddControllers();
-
-            services.AddSwaggerGen(x =>
-            {
-                x.SwaggerDoc("v1", new OpenApiInfo { Title = "MyTrip Api", Version = "v1" });
-            });
-        
-            var mappingConfig = new MapperConfiguration(mc =>
-            {
-              mc.AddProfile(new MappingProfile());
-            });
-
-            services.AddSingleton(mappingConfig.CreateMapper());
-        }
-    
-        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
-        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+    // This method gets called by the runtime. Use this method to add services to the container.
+    public void ConfigureServices(IServiceCollection services)
+    {
+
+      var frontConfiguration = Configuration.GetSection("Front").Get<FrontConfiguration>();
+      services.AddCors(options =>
+      {
+        options.AddPolicy(name: VueCorsPolicy,
+                                builder =>
+                                {
+                              builder
+                                      .AllowAnyHeader()
+                                      .AllowAnyMethod()
+                                      .AllowCredentials()
+                                      .WithOrigins(frontConfiguration.AddressFront);
+                            });
+      });
+
+      services.AddSingleton(frontConfiguration);
```

Comment:
why do we need singleton here?

Reply:
to skip this variable (this variable stores the address of the front), for example, to the password recovery service, so that you can then send the correct links, not static ones (for node errors.js is buggy and it has to be forcibly terminated, because of this, the port does not change)